appcues/src/main/java/com/appcues/debugger/DebuggerViewModel.kt

```diff
@@ -247,31 +241,42 @@ internal class DebuggerViewModel(
         return toMutableList().onEach { it.showOnFab = false }
     }
 
-    fun onDetailDismiss() {
-        // reset the view state to remove any deep link path
-        _uiState.value = Expanded(Debugger(null))
-    }
+    fun captureScreen(): Capture? =
+        AppcuesActivityMonitor.activity?.captureScreen(config, contextResources)
 }
 
 private fun Activity.captureScreen(
     config: AppcuesConfig,
     contextResources: ContextResources,
-) {
-    val timestamp = Date()
-    val view = window.decorView.rootView
-    val displayName = view.screenCaptureDisplayName(timestamp)
-    val layout = view.asCaptureView()
-    layout?.let {
-        val capture = Capture(
-            appId = config.applicationId,
-            displayName = displayName,
-            screenshotImageUrl = null,
-            layout = layout,
-            metadata = contextResources.generateCaptureMetadata(),
-            timestamp = timestamp,
-        )
+): Capture? =
+    window.decorView.rootView.let {
+        // hide the debugger view for screen capture, if present
+        val debuggerView = it.findViewById<View>(R.id.appcues_debugger_view)
+        debuggerView?.let { view ->
```

Comment:
we could in the same class have methods like prepare() and restore() for things that we need to do before and after